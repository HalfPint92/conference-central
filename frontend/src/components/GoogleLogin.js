import React from 'react';
import axios from 'axios';
import '../componentcss/LoginButtons.css';

const GoogleLogin = () => {
  // Function to handle Google login
  const handleGoogleLogin = async () => {
    try {
      // Send a GET request to the server to get the Google login URL.
      const { data } = await axios.get('http://localhost:5000/api/auth/google-login-url');

      // Redirect the user to the obtained Google login URL.
      // This will open the Google login page in a new tab or redirect the user to Google's login page.
      window.location.replace(data.url);
    } catch (error) {
      
      console.error(error);
    }
  };

  // The GoogleLogin component renders a button with Google's logo that triggers the Google login process when clicked.
  return (
    <button className="google-login-button" onClick={handleGoogleLogin}>
      
      <img src={`${process.env.PUBLIC_URL}/images/google-icon.jpg`} alt="Google Icon" />
      Sign in with Google
    </button>
  );
};

export default GoogleLogin;
