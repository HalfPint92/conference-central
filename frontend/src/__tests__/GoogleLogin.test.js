import React from 'react';
import GoogleLogin from '../components/GoogleLogin';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<GoogleLogin />).toJSON();
  expect(tree).toMatchSnapshot();
});


