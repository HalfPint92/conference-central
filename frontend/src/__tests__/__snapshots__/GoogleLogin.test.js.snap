// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="google-login-button"
  onClick={[Function]}
>
  <img
    alt="Google Icon"
    src="/images/google-icon.jpg"
  />
  Sign in with Google
</button>
`;
